docs/bugs.json

```diff
@@ -12,6 +12,19 @@
             "yulOptimizer": true
         }
     },
+    {
+        "uid": "SOL-2023-1",
+        "name": "MissingSideEffectsOnSelectorAccessBug",
+        "summary": "Accessing the ``.selector`` member on complex expressions leaves the expression unevaluated in the legacy code generation.",
+        "description": "When accessing the ``.selector`` member in an expression with side-effects like assignments, function calls or conditionals, the expression would not be evaluated in the legacy code generation. Note that the code generated by the IR pipeline was unaffected and will behave as expected.",
```

Comment:
```suggestion
        "description": "When accessing the ``.selector`` member on an expression with side-effects, like an assignment, a function call or a conditional, the expression would not be evaluated in the legacy code generation. This would happen in expressions where the functions used in the expression were all known at compilation time, regardless of whether the whole expression could be evaluated at compilation time or not. Note that the code generated by the IR pipeline was unaffected and would behave as expected.",
```